Userland/Libraries/LibFileSystemAccessClient/Client.h

```diff
@@ -18,8 +19,21 @@
 
 namespace FileSystemAccessClient {
 
+class Client;
+class File : public Core::Stream::File {
+public:
+    static ErrorOr<NonnullOwnPtr<File>> adopt_fd(Badge<Client>, String filename, int fd, Core::Stream::OpenMode);
+
+    String filename() const { return m_filename; }
+
+private:
+    File(String filename, Core::Stream::File);
+
+    String m_filename;
+};
+
 using DeprecatedResult = ErrorOr<NonnullRefPtr<Core::File>>;
-using Result = ErrorOr<NonnullOwnPtr<Core::Stream::File>>;
+using Result = ErrorOr<NonnullOwnPtr<File>>;
```

Comment:
I'm not so sure about this one tbh. It feels a bit hackish, but it's probably better than having a result type like this
```cc
struct FileAccessResult {
    NonnullOwnPtr<Core::Stream::File> file;
    String filename;
};
using Result = ErrorOr<FileAccessResult>;
```

Reply:
That's something that already have been discussed on discord:
https://discord.com/channels/830522505605283862/830739873119207426/1050376615361773588

Reply:
oh, thanks for the link. I will change the PR.